experimenter/experimenter/targeting/constants.py

```diff
@@ -134,6 +100,67 @@ def __post_init__(self):
 # June 26, 2025.
 ADS_DISABLED_BR_MX_2025_06_26 = TOPSITES_OR_SPONSORED_TOPSITES_DISABLED
 
+# User has at least one non-default privacy setting:
+#   ETP strict (see https://searchfox.org/firefox-main/source/browser/base/content/test/protectionsUI/browser_protectionsUI.js#193-200)
+#   Global Privacy Control (see https://searchfox.org/firefox-main/source/modules/libpref/init/StaticPrefList.yaml#16139-16145)
+#   Enable DNS over HTTPS - Max Protection (see https://firefox-source-docs.mozilla.org/networking/dns/dns-over-https-trr.html#implementation)
+#   Enable HTTPS-Only Mode in all windows (see https://searchfox.org/firefox-main/source/modules/libpref/init/StaticPrefList.yaml#4430-4432)
+
+HAS_PRIVACY_SETTING = """
+(
+    (
+        'browser.contentblocking.category'|preferenceValue) == "strict"
+        ||
+        'browser.contentblocking.category'|preferenceValue) == "custom"
+    )
+    ||
+    ('privacy.globalprivacycontrol.enabled'|preferenceValue)
+    ||
+    (
+        (
+            ('network.trr.mode'|preferenceValue) == 3
+            ||
+            ('doh-rollout.mode'|preferenceValue) == 3
+        )
+    ||
+    ('dom.security.https_only_mode'|preferenceValue)
+)
+"""
+# User has uBlock, Adblocker Ultimate, Adblock Plus, or Ghostery extensions
+HAS_AD_BLOCKER = """
+(
+    !!addonsInfo.addons
+    &&
+    (
+        !!addonsInfo.addons["uBlock0@raymondhill.net"]
+        ||
+        !!addonsInfo.addons["adblockultimate@adblockultimate.net"]
+        ||
+        !!addonsInfo.addons["d10d0bf8-f5b5-c8b4-a8b2-2b9879e08c5d"]
+        ||
+        !!addonsInfo.addons["jid1-NIfFY2CA8fy1tg@jetpack"]
+    )
+)
+"""
+
+TOU_EXPERIENCE_TOTAL = """
+(
+    (
+        (
+            (
+                '' + [
+                        ADS_DISABLED,
+                        HAS_PRIVACY_SETTING,
+                        HAS_AD_BLOCKER
+                    ]
+            ) | regExpMatch('true','g')
+        )
+        ||
+        []
+    ) | length
+)
+"""
```

Comment:
This trick with regexp capture groups is clever; it took me a bit to figure out what was going on, so I'm a bit concerned about future maintenance here without comments.